src/PerfView/EventViewer/EventWindow.xaml.cs

```diff
@@ -1108,20 +1112,64 @@ public bool Find(string pat)
                 }
 
                 var item = list[curPos] as EventRecord;
-                var foundItem = m_findPat.IsMatch(item.Rest) || m_findPat.IsMatch(item.EventName) ||
-                    m_findPat.IsMatch(item.ProcessName) || m_findPat.IsMatch(item.TimeStampRelatveMSec.ToString());
-                var fields = item.DisplayFields;
-                for (int i = 0; i < fields.Length; i++)
+                var foundItem = false;
+
+                // If the filter query tree is successfully generated, use it to find.
+                if (tree != null)
                 {
+                    Dictionary<string, string> data = new Dictionary<string, string>();
+                    data["ProcessName"] = item.ProcessName;
+                    data["TimestampRelativeMsec"] = item.TimeStampRelatveMSec.ToString();
+                    data["ProcessName"] = item.ProcessName;
+                    
+                    if (!string.IsNullOrEmpty(item.Rest))
+                    {
+                        foreach(var r in item.Rest.Split(FilterQueryUtilities.SpaceSeparator, StringSplitOptions.RemoveEmptyEntries))
+                        {
+                            // Format of Rest: Property0=Value0 Property1=Value1
+                            var splitOnEquals = r.Trim().Split('=');
+                            if (splitOnEquals.Length != 2)
+                            {
+                                continue;
+                            }
+
+                            data[splitOnEquals[0]] = splitOnEquals[1].Replace("\"", "");
+                        }
+                    }
+
+                    if (m_source.ColumnsToDisplay != null)
+                    {
+                        for(int displayFieldIdx = 0; displayFieldIdx < item.DisplayFields.Length; displayFieldIdx++)
+                        {
+                            data[m_source.ColumnsToDisplay[displayFieldIdx]] = item.DisplayFields[displayFieldIdx];
+                        }
+                    }
+
+                    foundItem = tree.Match(data, item.EventName);
```

Comment:
I'm wondering if it makes sense to pass the entire `EventRecord` in here and look for ways to optimize here, such as not parsing `Rest` until we have to.

Reply:
>> I'm wondering if it makes sense to pass the entire EventRecord in here

Definitely thought about that. I held back so far because: I figured I leave the input data type to be PerfView construct agnostic; specifically, the ``EventRecord`` type exists in the PerfView project but since I am building this functionality in the TraceEvent project I would either have to move the entire implementation from TraceEvent to PerfView or move the ``EventRecord`` definition to the TraceEvent library. 

My thinking here was that if there could be other uses of this filtering mechanism, might as well have it exposed as behavior in the TraceEvent side of things such as for cases where we'd possibly want to programmatically filter events outside Perfview. 

I'd be curious to know your opinion here and could definitely move things around. 

>>  such as not parsing Rest until we have to.

Definitely, the latest implementation takes care of this where we are only parsing the Rest column if the rest of the checks fail.